server/server.go

```diff
@@ -599,6 +599,10 @@ func validateOptions(o *Options) error {
 		return fmt.Errorf("lame duck grace period (%v) should be strictly lower than lame duck duration (%v)",
 			o.LameDuckGracePeriod, o.LameDuckDuration)
 	}
+	if int64(o.MaxPayload) > o.MaxPending {
```

Comment:
Should we add a warning for anything over say 8M? Just suggesting that is not recommended?

Reply:
Added in next commit: https://github.com/nats-io/nats-server/pull/2413/commits/d1365b74120c63e68c4bfcda336e97ad35a8991f

Reply:
Actually pushed other commit to use the constant instead of hardcoding in warning message.